datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -444,6 +437,24 @@ def reset_http_config(self):
         """
         self._http_handlers.clear()
 
+    def set_bearer_token(self, instance):
+        # Whether or not to use the service account bearer token for authentication.
+        # Can be explicitly set to true or false to send or not the bearer token.
+        # If set to the `tls_only` value, the bearer token will be sent only to https endpoints.
+        # If 'bearer_token_path' is not set, we use /var/run/secrets/kubernetes.io/serviceaccount/token
+        # as a default path to get the token.
+        if instance['bearer_token_auth'] == 'tls_only':
+            instance["bearer_token_auth"] = instance["prometheus_url"].startswith("https://")
+        else:
+            instance["bearer_token_auth"] = is_affirmative(instance["bearer_token_auth"])
+        
+        instance['bearer_token_path'] = instance.get('bearer_token_path', None)
+
+        instance['_bearer_token'] = self._get_bearer_token(instance['bearer_token_auth'], instance['bearer_token_path'])
+        instance['_bearer_token_last_refresh'] = time.time()
```

Comment:
We should probably remove these 3 lines from the __init__ function as well

Reply:
Good catch, thanks!